<?xml version="1.0"?>
<launch>
  <arg name="gui"        default="true"/>
<!--  <arg name="world_file" default="/home/sergi/catkin_ws/src/costmap_merge/worlds/world1.world"/>-->
  <arg name="world_file" default="$(env TURTLEBOT_GAZEBO_WORLD_FILE)"/>

  <arg name="base"       value="$(optenv TURTLEBOT_BASE kobuki)"/> <!-- create, roomba -->
  <arg name="battery"    value="$(optenv TURTLEBOT_BATTERY /proc/acpi/battery/BAT0)"/>  <!-- /proc/acpi/battery/BAT0 -->
  <arg name="stacks"     value="$(optenv TURTLEBOT_STACKS hexagons)"/>  <!-- circles, hexagons -->
  <arg name="3d_sensor"  value="$(optenv TURTLEBOT_3D_SENSOR kinect)"/>  <!-- kinect, asus_xtion_pro -->

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="use_sim_time" value="true"/>
    <arg name="debug" value="false"/>
    <arg name="gui" value="$(arg gui)" />
    <arg name="world_name" value="$(arg world_file)"/>
  </include>

  <group ns="robot1">
    <arg name="x" value="-5"/>
    <arg name="y" value="-3"/>
    <arg name="Y" value="0"/>
    <arg name="tf_prefix" value="robot1_tf"/>
    <param name="tf_prefix" value="$(arg tf_prefix)"/>
    <include file="$(find costmap_merge)/launch/includes/$(arg base).launch.xml">
      <arg name="x" value="$(arg x)"/>
      <arg name="y" value="$(arg y)"/>
      <arg name="Y" value="$(arg Y)"/>
      <arg name="base" value="$(arg base)"/>
      <arg name="stacks" value="$(arg stacks)"/>
      <arg name="3d_sensor" value="$(arg 3d_sensor)"/>
      <arg name="init_pose" value="-x $(arg x) -y $(arg y) -z 0 -R 0 -P 0 -Y $(arg Y)" />
      <arg name="robot_name"  value="robot1" />
      <arg name="tf_prefix" value="$(arg tf_prefix)"/>
      <arg name="robot_type" value="locator"/>
    </include>
  </group>

  <group ns="robot2">
    <arg name="x" value="0"/>
    <arg name="y" value="3"/>
    <arg name="Y" value="0"/>
    <arg name="tf_prefix" value="robot2_tf"/>
    <param name="tf_prefix" value="$(arg tf_prefix)"/>
    <include file="$(find costmap_merge)/launch/includes/$(arg base).launch.xml">
      <arg name="x" value="$(arg x)"/>
      <arg name="y" value="$(arg y)"/>
      <arg name="Y" value="$(arg Y)"/>
      <arg name="base" value="$(arg base)"/>
      <arg name="stacks" value="$(arg stacks)"/>
      <arg name="3d_sensor" value="$(arg 3d_sensor)"/>
      <arg name="init_pose" value="-x $(arg x) -y $(arg y) -z 0 -R 0 -P 0 -Y $(arg Y)" />
      <arg name="robot_name"  value="robot2" />
      <arg name="tf_prefix" value="$(arg tf_prefix)"/>
      <arg name="robot_type" value="pawn"/>
    </include>
  </group>

  <group ns="robot3">
    <arg name="x" value="-3"/>
    <arg name="y" value="0"/>
    <arg name="Y" value="3.1416"/>
    <arg name="tf_prefix" value="robot3_tf"/>
    <param name="tf_prefix" value="$(arg tf_prefix)"/>
    <include file="$(find costmap_merge)/launch/includes/$(arg base).launch.xml">
      <arg name="x" value="$(arg x)"/>
      <arg name="y" value="$(arg y)"/>
      <arg name="Y" value="$(arg Y)"/>
      <arg name="base" value="$(arg base)"/>
      <arg name="stacks" value="$(arg stacks)"/>
      <arg name="3d_sensor" value="$(arg 3d_sensor)"/>
      <arg name="init_pose" value="-x $(arg x) -y $(arg y) -z 0 -R 0 -P 0 -Y $(arg Y)" />
      <arg name="robot_name"  value="robot3" />
      <arg name="tf_prefix" value="$(arg tf_prefix)"/>
      <arg name="robot_type" value="pawn"/>
    </include>
  </group>
 
</launch>
